utils/kubernetes/kube_state_processor.py

```diff
@@ -72,6 +72,53 @@ def _extract_label_value(self, name, labels):
                 return label.value
         return None
 
+    # All these methods try to follow the same order as github.com/kubernetes/kube-state-metrics/README.md
+
+    # Labels attached: node, kernel_version, os_image, container_runtime_version, kubelet_version, kubeproxy_version
+    def kube_node_info(self, message, **kwargs):
+        """ Node information and versions. """
+        metric_name = NAMESPACE + '.node.info'
+        for metric in message.metric:
+            val = metric.gauge.value
+            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
+            self.gauge(metric_name, val, tags)
+
+    def kube_node_spec_unschedulable(self, message, **kwargs):
+        """ Whether a node can schedule new pods. """
+        metric_name = NAMESPACE + '.node.status'
+        statuses = ('schedulable', 'unschedulable')
+        for metric in message.metric:
+            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
+            status = statuses[int(metric.gauge.value)]  # value can be 0 or 1
+            tags.append('status:{}'.format(status))
+            self.gauge(metric_name, 1, tags)  # metric value is always one, value is on the tags
+
+    def kube_node_status_ready(self, message, **kwargs):
+        """ The ready status of a cluster node. """
+        service_check_name = NAMESPACE + '.node.ready'
+        for metric in message.metric:
+            name, val = self._eval_metric_condition(metric)
+            tags = ['node:{}'.format(self._extract_label_value("node", metric.label))]
+            if name == 'true' and val:
+                self.service_check(service_check_name, self.kube_check.OK, tags=tags)
+            elif name == 'false' and val:
+                self.service_check(service_check_name, self.kube_check.CRITICAL, tags=tags)
+            elif name == 'unknown' and val:
+                self.service_check(service_check_name, self.kube_check.UNKNOWN, tags=tags)
```

Comment:
I didn't write the original code, I'm just moving it around (check the left side in the GitHub files view).

I agree with you that it makes more sense that way, but perhaps there's a reason why things are done that way.